pkg/controller/backup_sync_controller.go

```diff
@@ -21,378 +21,408 @@ import (
-	kbClient                client.Client
-	podVolumeBackupClient   velerov1client.PodVolumeBackupsGetter
-	backupLister            velerov1listers.BackupLister
-	csiVSLister             snapshotv1listers.VolumeSnapshotLister
-	csiSnapshotClient       *snapshotterClientSet.Clientset
-	kubeClient              kubernetes.Interface
+type backupSyncReconciler struct {
+	client                  client.Client
 	namespace               string
-	defaultBackupLocation   string
 	defaultBackupSyncPeriod time.Duration
 	newPluginManager        func(logrus.FieldLogger) clientmgmt.Manager
 	backupStoreGetter       persistence.ObjectBackupStoreGetter
+	logger                  logrus.FieldLogger
 }
 
-func NewBackupSyncController(
-	backupClient velerov1client.BackupsGetter,
-	kbClient client.Client,
-	podVolumeBackupClient velerov1client.PodVolumeBackupsGetter,
-	backupLister velerov1listers.BackupLister,
-	csiVSLister snapshotv1listers.VolumeSnapshotLister,
-	syncPeriod time.Duration,
+// NewBackupSyncReconciler is used to generate BackupSync reconciler structure.
+func NewBackupSyncReconciler(
+	client client.Client,
 	namespace string,
-	csiSnapshotClient *snapshotterClientSet.Clientset,
-	kubeClient kubernetes.Interface,
-	defaultBackupLocation string,
+	defaultBackupSyncPeriod time.Duration,
 	newPluginManager func(logrus.FieldLogger) clientmgmt.Manager,
 	backupStoreGetter persistence.ObjectBackupStoreGetter,
-	logger logrus.FieldLogger,
-) Interface {
-	if syncPeriod <= 0 {
-		syncPeriod = time.Minute
-	}
-	logger.Infof("Backup sync period is %v", syncPeriod)
-
-	c := &backupSyncController{
-		genericController:       newGenericController(BackupSync, logger),
-		backupClient:            backupClient,
-		kbClient:                kbClient,
-		podVolumeBackupClient:   podVolumeBackupClient,
+	logger logrus.FieldLogger) *backupSyncReconciler {
+	return &backupSyncReconciler{
+		client:                  client,
 		namespace:               namespace,
-		defaultBackupLocation:   defaultBackupLocation,
-		defaultBackupSyncPeriod: syncPeriod,
-		backupLister:            backupLister,
-		csiVSLister:             csiVSLister,
-		csiSnapshotClient:       csiSnapshotClient,
-		kubeClient:              kubeClient,
-
-		// use variables to refer to these functions so they can be
-		// replaced with fakes for testing.
-		newPluginManager:  newPluginManager,
-		backupStoreGetter: backupStoreGetter,
+		defaultBackupSyncPeriod: defaultBackupSyncPeriod,
+		newPluginManager:        newPluginManager,
+		backupStoreGetter:       backupStoreGetter,
+		logger:                  logger,
 	}
+}
 
-	c.resyncFunc = c.run
-	c.resyncPeriod = 30 * time.Second
+// Reconcile syncs between the backups in cluster and backups metadata in object store.
+func (b *backupSyncReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := b.logger.WithField("controller", BackupSync)
+	log.Debug("Checking for existing backup storage locations to sync into cluster.")
```

Comment:
Seems this log message isn't reasonable anymore

Reply:
Modified.